File: win-prog/ProgWinCS/Hello Windows Forms/PaintEvent/PaintEvent.cs
//-----------------------------------------
// PaintEvent.cs © 2001 by Charles Petzold
//-----------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class PaintEvent
{
     public static void Main()
     {
          Form form   = new Form();
          form.Text   = "Paint Event";
          form.Paint += new PaintEventHandler(MyPaintHandler);

          Application.Run(form);
     }
     static void MyPaintHandler(object objSender, PaintEventArgs pea)
     {
          Graphics grfx = pea.Graphics;

          grfx.Clear(Color.Chocolate);
     }
}

File: win-prog/ProgWinVB/Hello Windows Forms/PaintEvent/PaintEvent.vb
'-------------------------------------------
' PaintEvent.vb (c) 2002 by Charles Petzold
'-------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Module PaintEvent
    Sub Main()
        Dim frm As New Form()
        frm.Text = "Paint Event"
        AddHandler frm.Paint, AddressOf MyPaintHandler

        Application.Run(frm)
    End Sub

    Sub MyPaintHandler(ByVal obj As Object, ByVal pea As PaintEventArgs)
        Dim grfx As Graphics = pea.Graphics
        grfx.Clear(Color.Chocolate)
    End Sub
End Module
